examples/component-example.py

```diff
@@ -0,0 +1,122 @@
+import pygfx as gfx
+from wgpu.gui.auto import run, WgpuCanvas
+
+from collagraph import Collagraph, Component, create_element as h, EventLoopType
+from collagraph.renderers import PygfxRenderer
+
+
+class Track(Component):
+    track_geometry = gfx.cylinder_geometry()
+    track_material = gfx.MeshPhongMaterial(color=[1, 0.5, 1])
+
+    def render(self):
+        return h(
+            "Mesh",
+            {
+                **self.props,
+                "geometry": Track.track_geometry,
+                "material": Track.track_material,
+            },
+        )
+
+
+class Scrubber(Component):
+    scrubber_geometry = gfx.sphere_geometry()
+    scrubber_material = gfx.MeshPhongMaterial(color=[0.5, 1, 0.5])
+
+    def __init__(self, props):
+        super().__init__(props)
+        self._captured = False
+        self._sphere = None
+        self._mouse_pos = (0, 0)
+        self.position = [0, 0, 0]
+
+    def mouse_down(self, event):
+        if not self._sphere:
+            self._sphere = event.current_target
```

Comment:
the challenge here is getting a reference to the instantiated sphere Mesh, correct? love how this is resolved so cleanly 👍

Reply:
Hehe, well, it still feels kind of hacky... I do think we'll want to implement something like Vues' `$el` and `$refs` functionality, but not for this PR yet ;)

Reply:
In a way, they would just do the same kind of thing but just in response to a different event (mounted). So, hacky, maybe. I would say it's just less convenient. :)